### DISCUSSION TITLE
Help me with my workflow and dependencies?

### DISCUSSION THREAD

**Discussion by KjellConnelly** [nanojsx/nano#73] [Help / Questions] [answered]
Hello! I was hoping you could give me some advice on how to get started. You see, I have a nodejs Express server, which usually is used to run api calls. But occasionally, I send some raw html. Typically something like:

```javascript
let html = `
<html>
<h2>Welcome</h2>
<input id="input" />
<button onclick="let input = document.getElementById('input');fetch('website/api/signup', {....">
   Sign Up
</button>
</html>`
res.send(html);
```

It's pretty bad, and I'm used to using react. But I also am running my server on a low capacity computer and don't want to use a bunch of resources either building, or having a lot of dependencies. I would really like to work with JSX. Escaping strings is a nightmare.

Anyways, I've tried your SSR example, but I don't want to use typescript, so I was running it for testing like `node index.js`. And of course, it doesn't understand JSX. So I tried with Tagged Templates, which is nice, but not nice enough. I'm still escaping stuff. 

Do you have any ideas on how to minimize the number of dependencies and work my computer has to do to just generate some simple reactive html that is in JSX format? I've tried `@babel/node` with `@babel/preset-react`, but this kinda defeats the purpose of nano being that it's requiring React now with a virtual dom, which is not what I want.

**Comment by jrson83** [comment]
To run Nano JSX with SSR you only need a few packages. Have a look at the [example1 package.json](https://github.com/nanojsx/example1/blob/main/package.json). This is the minimal setup. Actually you can even reduce the size of your bundle.js when you import only single components like:
```JSX
import * as Nano from 'nano-jsx/lib/core'
import { Component } from 'nano-jsx/lib/component'
import { Helmet } from 'nano-jsx/lib/components/helmet'
import * as Router from 'nano-jsx/lib/components/router'
```
Besides I recommend using the [tsconfig.json](https://github.com/nanojsx/template/blob/master/tsconfig.json) settings from the template.

I also come from react and I just can recommend dropping the usage completly. The usage of Nano JSX compared to react is almost the same and within some days you can learn how to use all components.

As far as I can see in your code has some errors, since you assign your html code to the `let s` but you send the variable `html`  which seems like not defined. Checkout the [exampel1 server.tsx](https://github.com/nanojsx/example1/blob/main/src/server.tsx) for usage.

**Comment by yandeu** [comment, marked as answer]
I just made a simple example for you.
See: [https://github.com/nanojsx/node-jsx-example](https://github.com/nanojsx/node-jsx-example).

**Comment by KjellConnelly** [comment]
Thank you two! I've integrated some the new `node-jsx-example`. I really appreciate the quick response and thorough help! I didn't know typescript was its own compiler. Guess that's because the only times I've used it, I was using rollup which was more of a copy/paste solution to my other projects. I got stuck last night updating my raspberry pi (it still had node 8.x installed, and had up update a ton of things just to get the server running again). On my mac, my test project works now just generating the .jsx to .js files and `renderSSR()` function to generate the html. Looks great!

@jrson83 Yep, that example code was a quick writeup of what I was basically doing. It wasn't actual code from my project (code is abstracted a bit more) hence why I was sending `html` instead of `s`. Lol, whoops!

I'll comment again when I get things working on the pi how I intend it to, just as an update. Probably be a few days before I can get back to messing with it. Ttyl!

**Reply by yandeu** [reply]
This should help even more :blush: 
https://www.youtube.com/playlist?list=PLC2Z8IWl1XDJI4Ah7ABolQ79AugF_eH1g

**Reply by jrson83** [reply]
Thanks! I didn't know that tutorials exist! Damn youtube search algorithm.

**Reply by yandeu** [reply]
I made it yesterday and uploaded it few hours ago :laughing:
